src/Rules/DeadCode/PossiblyPureStaticCallCollector.php

```diff
@@ -0,0 +1,55 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\DeadCode;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Expression;
+use PHPStan\Analyser\Scope;
+use PHPStan\Collectors\Collector;
+
+/**
+ * @implements Collector<Node\Stmt\Expression, array{class-string, string, int}>
+ */
+class PossiblyPureStaticCallCollector implements Collector
+{
+
+	public function __construct()
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return Expression::class;
+	}
+
+	public function processNode(Node $node, Scope $scope)
+	{
+		if (!$node->expr instanceof Node\Expr\StaticCall) {
+			return null;
+		}
+		if (!$node->expr->name instanceof Node\Identifier) {
+			return null;
+		}
+
+		if (!$node->expr->class instanceof Node\Name) {
+			return null;
+		}
+
+		$methodName = $node->expr->name->toString();
+		$calledOnType = $scope->resolveTypeByName($node->expr->class);
+		$methodReflection = $scope->getMethodReflection($calledOnType, $methodName);
```

Comment:
We don't need to go through the type system at all. We have a class name and a method name as strings at hand, so we should go through ReflectionProvider and `ClassReflection::getNativeMethod()` instead. Thanks.

Reply:
Oh scratch that, this actually avoids some other problems.